src/ui/server/helpers/has-form-data/index.test.ts

```diff
@@ -0,0 +1,26 @@
+import hasFormData from '.';
+
+describe('helpers/has-form-data', () => {
+  describe('when a form has a field that is not _csrf', () => {
+    it('should return true', () => {
+      const mockFormData = {
+        _csrf: '1234',
+        mockField: true,
+      };
+
+      const result = hasFormData(mockFormData);
+
+      expect(result).toEqual(true);
+    });
+  });
+
+  describe('when a form does NOT have a field other than _csrf', () => {
+    it('should return false', () => {
```

Comment:
Do we need to add a coverage where the form field is empty or void?

```
  const mockFormData = {
        _csrf: '1234',
        mockField,
      };

Reply:
Excellent point @abhi-markan yes need to improve this it thinks this is "a form with data"

```js
Object.keys({ test: '' }).length
Object.keys({ test: null }).length
Object.keys({ test: undefined }).length
```

...Which is technically true 😉 I'll add some checks to account for this

Reply:
Added a function to handle this, `stripEmptyFormFields` Great spot 👍